RELEASE.md

```diff
@@ -0,0 +1,63 @@
+# Release
+
+The processs of creating a release is as follows.
+
+## Schedule
+
+Releases are made on an ad-hoc basis currently, rather than on an
+elapsed time basis.  Typically once there are significant changes.
+
+## Invariants / rules
+
+- Pull Requests should not change the version number in `project.clj`.
+- Each Pull Request should update the `CHANGELOG.md` file as
+  appropriate.
+- In the normal state of this repository, the version in `project.clj`
+  should be of the form `x.y.z-SNAPSHOT`, denoting that the repo is in
+  between versions.
+
+## Release preconditions
+
+- Your local repo should be clean.
+- All tests pass.
+
+## Release process
+
+Note this uses `0.1.12` as an example new release number.
+
+1. Create a release branch on your local repo of your fork of this
+  repo on Github.
+  ```
+  git checkout master
+  git pull
+  git checkout -b release-0.1.12
+  git push -u origin release-0.1.12
+  ```
+2. Update [CHANGELOG.md](CHANGELOG.md)
+  * Write the release highlights in the `Unreleased` section.
+  * Add a section for the new release just under the `Unreleased` section.
+  * Update the links section at the end of the file to include links
+    for the new release and update the link for the Unreleased
+    version.
+3. Update the version in [project.clj](project.clj). 
+4. Update any version numbers in [README.md](README.md).
+5. Push changes to your fork on Github.
+  ```
+  git add .
+  git commit -m "Release 0.1.12"
+  git push origin
+  ```
+6. Create a pull request. 
+7. Get other maintainers to review the pull request.
+8. If changes are required, update the pull request.
+9. Once all is okay, merge the pull request into the primary repo on
+   Github.
+10. Tag the new commit on Github with the new release number (e.g.,
+    `0.1.12`).
+11. Create a new release on Github with the new tag and the same
```

Comment:
I don't think we need to bother with GH releases since our primary distribution is via Clojars. When you create a tag, it will be visible on the GH releases page automatically I believe so that perhaps satisfy those that are, for some reasons, interested in that.

Reply:
We can hold off for now, but I do think that it is nice for people who browse Github to see what releases there have been.

Reply:
As mentioned, I believe tags will do that for us.

Reply:
Agreed.  I have removed the parts about making a Github release.